src/twisted/names/authority.py

```diff
@@ -312,7 +312,7 @@ def loadFile(self, filename):
         fp = FilePath(filename)
         # Not the best way to set an origin. It can be set using $ORIGIN
         # though.
-        self.origin = nativeString(fp.basename() + b".")
+        self.origin = nativeString(fp.basename()) + "."
```

Comment:
Do we still need to use `nativeString` here ?

I think that this was used only for py2 compatibility.

We now support Python 3 only.

Reply:
Yep, we should be working to remove any existing uses of `nativeString` and not adding new ones.

Reply:
Given that `nativeString` always returns a `str` now, this was just broken if `filename` was a `bytes` object, which suggests that there's still a missing test if you want the signature to be `bytes | str`.

```suggestion
        self.origin = fp.asTextMode().basename() + "."
```